packages/legend-application-query/src/components/QueryEditor.tsx

```diff
@@ -190,11 +201,216 @@ const renderQueryEditorHeaderLabel = (
   return null;
 };
 
+const QueryLoader = observer(
+  (props: {
+    editorStore: QueryEditorStore;
+    applicationStore: ApplicationStore<
+      LegendQueryApplicationConfig,
+      LegendQueryApplicationPlugin
+    >;
+  }) => {
+    const { editorStore, applicationStore } = props;
+    const queryFinderRef = useRef<SelectComponent>(null);
+    const [selectedQueryID, setSelectedQueryID] = useState('');
+    const [isMineOnly, setIsMineOnly] = useState(false);
+    const [searchText, setSearchText] = useState('');
+    const closeQueryImporter = (): void => {
+      editorStore.setIsQueryLoaderOpen(false);
+    };
+    const handleEnterQueryImporter = (): void =>
+      queryFinderRef.current?.focus();
+    const loadSelectedQuery = (): void => {
+      if (selectedQueryID) {
+        editorStore.setIsQueryLoaderOpen(false);
+        applicationStore.navigator.goTo(
+          generateExistingQueryEditorRoute(selectedQueryID),
+        );
+        window.location.reload();
+        applicationStore.setBlockingAlert({
+          message: 'Loading query...',
+          showLoading: true,
+        });
+      } else {
+        applicationStore.notifyWarning(
+          'No query is selected. Please select a query to load or close.',
+        );
+      }
+    };
+    const searchInputRef = useRef<HTMLInputElement>(null);
```

Comment:
see what we do in `EditExistingQuerySetup` and please bring over the `debounce` logic as well as `useEffect()` logic. We cannot search on every keystroke user makes and also, in the very beginning, we would like to already show some queries